iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -177,12 +177,9 @@ CombinedScanTask> applyConfig(
     }
     String schemaStr = conf.get(InputFormatConfig.READ_SCHEMA);
     if (schemaStr != null) {
-      scan.project(SchemaParser.fromJson(schemaStr));
-    }
-
-    String[] selectedColumns = conf.getStrings(InputFormatConfig.SELECTED_COLUMNS);
-    if (selectedColumns != null) {
-      scan.select(selectedColumns);
+      scan = scan.project(SchemaParser.fromJson(schemaStr));
```

Comment:
why don't we use helper methods:
````
Schema readSchema = InputFormatConfig.readSchema(conf);
if (readSchema != null) {
  scan = scan.project(readSchema)
} else {
  String[] selectedColumns = InputFormatConfig.selectedColumns(conf);
  if (selectedColumns != null) {
    scan = scan.select(selectedColumns)
  }
}
````

Reply:
btw, there is `expectedSchema = readSchema(conf, table, caseSensitive)`
but it creates a Hive schema with virtual columns, maybe that could be reused

Reply:
> btw, there is `expectedSchema = readSchema(conf, table, caseSensitive)` but it creates a Hive schema with virtual columns, maybe that could be reused

I didn't find a  better way to reuse this method. :(